rcl/src/rcl/common.h

```diff
@@ -23,16 +23,6 @@ extern "C"
 #include "rcl/error_handling.h"
 #include "rcl/types.h"
 
-#define RCL_CHECK_ARGUMENT_FOR_NULL(argument, error_return_type, allocator) \
```

Comment:
For my understanding, is it intentional that the files that use `RCL_CHECK_ARGUMENT_FOR_NULL` can only do so because it's implicitly included in `common.h`? FWIU we have an "include what you use" policy that might suggest those files (e.g. `client.c`) should now include `rcl/error_handling.h` explicitly, but the purpose of `common.h` might be to get around that rule in some circumstances.

Reply:
I don't think that was the point of common.h I removed the include from common.h and filled in every source file that had an issue, then added it back so common.c could compile and link correctly.

Reply:
it should be fine to remove it from common.h now right? or could you give more info on why it's still needed?

Reply:
I believe it needs the types as it returns some `rcl_ret_t`

Reply:
they should come from rcl/types.h no?

Reply:
yes exactly, all I meant is currently common.c doesnt include types.h

Reply:
Looks like I misunderstood what we're talking about here. If now all files include `rcl/error_handling.h` (including `common.c`) there should indeed be no need to include it in `common.h`. Not sure what the linking issue you faced was from

Reply:
The linking issue was due to one more missing include. I had to trace back to an `implicit declaration of function` that was caused by the missing macro. I cleared the include from `common.h` in 
a5e5825